lib/timers/events.rb

```diff
@@ -60,18 +60,12 @@ def schedule(time, callback)
       # Maintain sorted order, O(logN) insertion time.
       @sequence.insert(index, handle)
       
-      return handle
+      handle
     end
     
     # Returns the first non-cancelled handle.
     def first
-      while handle = @sequence.last
-        if handle.cancelled?
-          @sequence.pop
-        else
-          return handle
-        end
-      end
+      @sequence.reverse.find { |handle| !handle.cancelled? }
```

Comment:
For now I will revert the old version but keep the new version there, commented out. I'm not as deeply familiar with `Timers` as everything else, especially after the refactor; so I will come back to this when I find myself back in this edge of the Celluverse.